src/string.c

```diff
@@ -46,3 +47,47 @@ longlong xcp_str_to_longlong (const char *str, bool *ok) {
 
   return value;
 }
+
+// -----------------------------------------------------------------------------
+
+static inline char *xcp_create_hex_buf (const void *buf, size_t count, bool reverse) {
+  if (!count) return NULL;
+
+  static const char prefix[] = "0x";
+  static const size_t prefixSize = sizeof prefix - 1;
+
+  const size_t hexBufSize = prefixSize + count * 2 + 1;
+  char *hexBuf = malloc(hexBufSize);
+  if (!hexBuf) return NULL;
```

Comment:
Log an error maybe ?

Reply:
Bad idea, it's a util library. I don't know the user usage, it can be a daemon or simple CLI program, in the two cases the log system is not the same.